src/lib/y2storage/setup_checker.rb

```diff
@@ -78,7 +80,32 @@ def errors
     #
     # @return [Array<SetupError>]
     def warnings
-      boot_warnings + product_warnings + mount_warnings + security_policy_warnings
+      boot_warnings + product_warnings + mount_warnings + security_policy_warnings + encryption_warnings
+    end
+
+    # All encryption warnings detected in the setup
+    #
+    # Argion2* needs at least 4GByte momory. cryptsetup could crash with less than 4GByte
+    # RAM.(bsc#1246876)
+    #
+    # @return [Array<SetupError>]
+    def encryption_warnings
+      return [] if Yast2::HwDetection.memory >= ((4000 - 64) << 20)
```

Comment:
Please, use DiskSize:


```suggestion
      return [] if DiskSize.new(StorageManager.instance.arch.ram_size) >= DiskSize.GiB(4)
```

Reply:
Wow, that's nice !!!!